src/coreclr/nativeaot/System.Private.CoreLib/src/System.Private.CoreLib.csproj

```diff
@@ -347,6 +352,8 @@
     <Compile Include="System\Diagnostics\StackFrame.NativeAot.Wasm.cs" />
     <Compile Include="System\Diagnostics\StackFrame.NativeAot.Wasi.cs" Condition="'$(TargetsWasi)' == 'true'" />
     <Compile Include="System\Diagnostics\StackFrame.NativeAot.Browser.cs" Condition="'$(TargetsBrowser)' == 'true'" />
+    <Compile Include="System\Threading\PortableThreadPool.NativeAot.Browser.cs" Condition="'$(TargetsBrowser)' == 'true' and '$(WasmEnableThreads)' == 'true'" />
+    <Compile Include="System\Threading\ThreadPool.NativeAot.Browser.cs" Condition="'$(TargetsBrowser)' == 'true' and '$(WasmEnableThreads)' == 'true'" />
```

Comment:
Why are the portable thread pool files insufficient and we need these as well?

Reply:
The problem is this line
https://github.com/dotnet/runtimelab/blob/79c861e8e1abb8e86b65ebc74b8723c135dd66cf/src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolWorkQueue.cs#L1169
As `YieldFromDispatchLoop` is internal perhaps it should really be in `PortableThreadPool`, but that would be a larger question.

Reply:
I think we should follow the pattern of stealing Mono files for these (as below). <https://github.com/dotnet/runtime/blob/d817f032075f77c7852b9f653c064a3346948667/src/mono/System.Private.CoreLib/src/System/Threading/ThreadPool.Browser.Threads.Mono.cs#L10> for `YieldFromDispatchLoop`, <https://github.com/dotnet/runtime/blob/d817f032075f77c7852b9f653c064a3346948667/src/mono/System.Private.CoreLib/src/System/Threading/PortableThreadPool.Browser.Threads.Mono.cs> for the other two properties.

Reply:
You can upstream fixes to move shareable files under libraries.

Prior art: https://github.com/dotnet/runtime/pull/114548/